components/tag/TagsInput.js

```diff
@@ -1,20 +1,32 @@
 import { useRef } from "react";
 import XMarkIcon from "@heroicons/react/20/solid/XMarkIcon";
 import Input from "../form/Input";
-
-export default function TagsInput({ tags, onTagAdd, onTagRemove }) {
+import Notification from "@components/Notification";
+export default function TagsInput({ tags, onTagAdd, onTagRemove ,showNotification,setShowNotification}) {
   const inputRef = useRef(null);
-
+  
   //key code
   const { comma, backspace } = {
     comma: 188,
     backspace: 8,
   };
 
+
+  const maxTagChar = 32; 
   const handleKeyUp = (e) => {
     const inputValue = inputRef.current.value;
-    if (e.keyCode === comma || inputValue.endsWith(",")) {
-      const newTag = inputValue.trim().replace(/,/g, "");
+    if(inputValue.length >= maxTagChar && e.keyCode !== backspace){  // '=' sign because in 32 char ',' sign does't count
+      setShowNotification({
+        show: true,
+      type: "error",
+      message: "Tag",
```

Comment:
We can say "Tag LImit Exceeded" in the message?

Reply:
Yeah it would be more expressive and helpful

Reply:
ok